### PR TITLE
CORE-802: Add PR template

### PR DESCRIPTION
**Pull request by jdeitchman**

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+### Instructions for Folks Outside of Core Creating PRs
+- Please add at least one Core engineer as a reviewer
+- Be sure to fill in the Purpose, Impact, and Priority sections above
+- Ping the Core team in Slack (you can use @-core in #collab-analyze-core, #collab-reply-core, or #prod-core depending on what team you're in) to give a heads up that the PR is ready for review.  If the change is urgent/critical, include this context.
+- Wait for a Core engineer to get back to you.  If the scope of the changes feels small and the risk feels low, the Core team may give you the green light to go ahead and merge the changes now and we'll circle back with the review afterwards.  If the changes feel more risky or have a broader impact, we may ask that you wait to merge the changes until someone from Core has had a chance to review the changes.
+- After your changes have been merged and deployed, please monitor Bugsnag to see if any new errors are reported. For session-manager the Bugsnag dashboard for errors introduced in the last 24hrs can be viewed [here](https://app.bugsnag.com/buffer/core-session-service/errors?filters[event.since][0]=30d&filters[error.status][0]=new&sort=last_seen).
```
Comment (jdeitchman) [inline pull_request_template.md]: Heya @philippemiguet!  For the session-manager library does it make sense to reference the session-service Bugsnag dashboard?

**Comment by esclapes** [comment]
Hey @jdeitchman thanks for kick-starting this, it feels to me that templates will help us manage the collaborations from other teams. Great idea.

Regarding this specific one in the `session-manager`, I just a couple of observations:

* this is used as an npm package, so a new version needs to be pushed and installed in a project before it has any direct impact. Would a roll-out plan be a good addition to this template?
* it will usually require a change in the `session-service` unless it is a bug fix or _client-only_ feature that can be supported with the existing service. I think identifying what changes are required the `session-service` and what is the related PR to that would also be a good addition.

Keen to get this ball rolling :raised_hands:

File: pull_request_template.md (hunk 1 above)
Comment (philippemiguet) [reply pull_request_template.md]: This library is intended to be used in other repository to talk with the Session service. Similar to the micro-rpc and buffer-rpc library, I would not specify any Bugsnag link here, and when the lib is used in another repo, the PR template will hint engineers to check Bugsnag for the app.

Comment (philippemiguet) [inline pull_request_template.md]:
```suggestion
- After your changes have been merged and deployed, please monitor Bugsnag to see if any new errors are reported.
```

**Pull request merged**